src/database.h

```diff
@@ -30,6 +30,12 @@ class Database
 		return instance;
 	}
 
+	static Database& getAsyncInstance()
+	{
+		static Database asyncInstance;
+		return asyncInstance;
+	}
```

Comment:
IIRC `getInstance` already returns an "async" instance, used for other purposes but should be suitable. Why having another instance?

On that note, singleton sucks, but a "doubleton" is even worse 😆 let's just make it not a global and pass it around

Reply:
I agree that "doubleton" sux and also was actually about using databaseTasks but we need to run each save which consists of multiple queries in single transaction which is currently not supported by databaseTasks, it's only single query per task. So we need kind of QuerySet which can be pushed to databaseTasks and run in single transaction. What's the proposed approach?